### PR TITLE
Wagtail 4.2 support

### PR DESCRIPTION
**Pull request by katdom13**
- Dropped support for Wagtail versions < 4.1
- Code cleanup
- Updated documentations

All tests run OK

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,9 +1,7 @@
 [tox]
 envlist =
-    py{37,38,39,310}-dj{30,31,32}-wagtail215
-    py{39,310}-dj{32,40}-wagtail216
-    py3{39,310}-dj{32,40}-wagtail{30,40}
-    py3{39,310}-dj{41}-wagtail{40,main}
+    py{37}-dj{32}-wagtail{41,42,main}
+    py{38,39,310,311}-dj{32,40,41}-wagtail{41,42,main}
```
Comment (nickmoreton) [inline tox.ini:4]:
```suggestion
    py{38,39,310}-dj{32,40,41}-wagtail{41,42}
    py{311}-dj{41}-wagtail{41,42,main}
```

File: tox.ini [hunk 2]
```diff
@@ -13,15 +11,11 @@ setenv =
...
-    dj31: Django>=3.1,<3.2
     dj32: Django>=3.2,<4.0
     dj40: Django>=4.0,<4.1
     dj41: Django>=4.1,<5.0
-    wagtail215: wagtail>=2.15,<2.16
-    wagtail216: wagtail>=2.16,<2.17
-    wagtail30: wagtail>=3.0,<4.0
-    wagtail40: wagtail>=4.0,<5.0
+    wagtail41: wagtail>=4.1,<4.2
+    wagtail42: wagtail>=4.2,<4.3
```
Comment (nickmoreton) [inline tox.ini:18]:
```suggestion
    wagtail42: wagtail>=4.2,<5.0
```

**Review by nickmoreton — COMMENTED** [review]
I think it would be good to use the [tox guthub actions](https://github.com/ymyzk/tox-gh-actions) here so the actions script is less complicated.

File: tox.ini (hunk 1 above)
Comment (katdom13) [reply tox.ini]: Applied suggestion
https://github.com/torchbox/wagtail-storages/pull/36/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R3-R5

File: tox.ini (hunk 2 above)
Comment (katdom13) [reply tox.ini]: Applied suggestion
https://github.com/torchbox/wagtail-storages/pull/36/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R27

**Comment by katdom13** [comment]
Hi @nickmoreton ,
Thank you for reviewing. I used the tox-gh-actions for the test workflows and it's easier to work with.

**Review by nickmoreton — APPROVED** [review]
Looking good thanks @katdom13

**Pull request merged**